torch/nn/modules/module.py

```diff
@@ -446,6 +446,7 @@ def forward(self, x):
     _modules: Dict[str, Optional['Module']]
     call_super_init: bool = False
     _compiled_call_impl : Optional[Callable] = None
+    ___needs_generation_tag_patch: bool  # Set by `install_generation_tagging_init`
```

Comment:
I don't see why we would need to add this here though. Many people add random attributes to the Module. We can't add all of them to the base class.

Reply:
Could you explain a bit more? As I understand, we do want to make the codebase as statically typed as possible. If there's an attribute being used in the code, why would we not want it to be declared explicitly, as opposed to inserting `#type: ignore` comments everywhere? Is the concern that people will take type annotations as promising some sort of API contract that we don't want to maintain?

Reply:
> As I understand, we do want to make the codebase as statically typed as possible.

The problem is that the codebase isn't static . Adding this annotation promisses that this attribute exists, but it does not! So this is not really fixing your problem (even though it silences the linter).

And yes this is a slippery slope as many features as implemented by adding attributes to Module. For example fsdp: https://github.com/pytorch/pytorch/blob/79e3833703c421cb9cb9ec2c45751a09abbe6e9b/torch/distributed/_composable/contract.py#L197
We cannot realistically start adding static types in the base Module for all of these.

Note that none of the parameters or buffers being registered are typed either.

Reply:
One other alternative to the `type-ignore` comments is to add a mixin that declares `__getattr__` and `__setattr__` methods which just dispatch directly to Python's own implementation of those methods. Then mypy will not warn on any attribute access for instances of that class (since in theory those methods could have other implementations).

Reply:
@albanD to be honest, I don't see why we shouldn't add the type annotations for everything. How else are you going to make sure they don't conflict